test/Serilog.Tests/Core/LoggerTests.cs

```diff
@@ -116,5 +119,266 @@ public void PropertiesCanBeBound()
             Assert.Equal("Name", property.Name);
             Assert.Equal("World", property.Value.LiteralValue());
         }
-    }
+
+		[Fact]
+		public void VerboseMethodsMatchConvention()
+		{
+			ValidateConventionForMethodSet("Verbose");
+		}
+
+		[Fact]
+		public void DebugMethodsMatchConvention()
+		{
+			ValidateConventionForMethodSet("Debug");
+		}
+
+		[Fact]
+		public void InformationMethodsMatchConvention()
+		{
+			ValidateConventionForMethodSet("Information");
+		}
+
+		[Fact]
+		public void WarningMethodsMatchConvention()
+		{
+			ValidateConventionForMethodSet("Warning");
+		}
+
+		[Fact]
+		public void ErrorMethodsMatchConvention()
+		{
+			ValidateConventionForMethodSet("Error");
+		}
+
+		[Fact]
+		public void FatalMethodsMatchConvention()
+		{
+			ValidateConventionForMethodSet("Fatal");
+		}
+
+		private void ValidateConventionForMethodSet(string setName)
+		{
+			var methodSet = typeof(Logger).GetMethods().Where(method => method.Name == setName);
+
+			var testMethods = typeof(LoggerTests).GetRuntimeMethods()
+				.Where(method => Regex.IsMatch(method.Name, "ValidateMethod\\d"));
+
+			foreach (var method in methodSet)
+			{
+				Assert.Equal(method.ReturnType, typeof(void));
+
+				Assert.True(method.IsPublic);
+
+				var messageTemplateAttr = method.GetCustomAttribute<MessageTemplateFormatMethodAttribute>();
+
+				Assert.NotNull(messageTemplateAttr);
+
+				Assert.Equal(messageTemplateAttr.MessageTemplateParameterName, "messageTemplate");
+
+				var signatureMatch = false;
+
+				foreach (var test in testMethods)
+				{
+					try
+					{
+						test.Invoke(this, new object[] { method });
+
+						signatureMatch = true;
+
+						break;
+					}
+					catch (Exception e)
+					when (e is TargetInvocationException
+							&& ((TargetInvocationException)e).GetBaseException() is XunitException)
+					{
+						continue;
```

Comment:
But still leaves the case of determining if an Assertion that is not in the Signature test fails... I should have a try catch statement in it, that adds a boolean value to the exception. It can be used to discriminate between signature test and invocation assertion failures.